tests/microos/networkmanager.pm

```diff
@@ -0,0 +1,130 @@
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use transactional;
+
+# the below network confiration is used if we use qemu user net
+my $dhcp_server = '10.0.2.2';
+my $dns_server = '10.0.2.3';
+my $mac_addr;
+my $nic_name;
+my $local_ip = '10.0.2.15';
+
+sub ping_check {
+    $mac_addr = get_var('NICMAC');
+    $nic_name = script_output("grep $mac_addr /sys/class/net/*/address |cut -d / -f 5");
+    assert_script_run("ping -c 5 $dhcp_server");
+    assert_script_run("ping -c 5 $dns_server");
+    # disconnect the device
+    assert_script_run("nmcli device disconnect $nic_name");
+    if (script_run("ping -c 5 $dns_server") == 0) {
+        die('The network is still up after disconnection');
+    }
+    assert_script_run("nmcli device connect $nic_name");
+    assert_script_run("ping -c 5 $dhcp_server");
+    assert_script_run("ping -c 5 $dns_server");
+}
+
+# double check what DNS-Manager is currently used by NetworkManger
+sub dns_mgr {
+    my $RcManager
+      = script_output(
+'dbus-send --system --print-reply --dest=org.freedesktop.NetworkManager /org/freedesktop/NetworkManager/DnsManager org.freedesktop.DBus.Properties.Get string:org.freedesktop.NetworkManager.DnsManager string:RcManager'
+      );
+    my $mode = script_output(
+'dbus-send --system --print-reply --dest=org.freedesktop.NetworkManager /org/freedesktop/NetworkManager/DnsManager org.freedesktop.DBus.Properties.Get string:org.freedesktop.NetworkManager.DnsManager string:Mode
+        '
+    );
+    return ($RcManager, $mode);
+}
+
+sub run {
+    # make sure 'sysconfig' and 'sysconfig-netconfig' are not installed by default
+    my @pkgs = ('sysconfig', 'sysconfig-netconfig');
+    foreach my $pkg (@pkgs) {
+        die "$pkg will not be installed by default on ALP" if (script_run("rpm -q $pkg") == 0);
+    }
+    my ($RcManager, $mode);
+    $mac_addr = get_var('NICMAC');
+    $nic_name = script_output("grep $mac_addr /sys/class/net/*/address |cut -d / -f 5");
+    # check 'NetworkManager' service is up and it can get right DNS server
+    systemctl('is-active NetworkManager');
+    assert_script_run('grep "Generated by NetworkManager" /etc/resolv.conf');
+    assert_script_run qq(grep "nameserver $dns_server" /etc/resolv.conf);
+    # DNS-Manager check
+    record_info('default dns bind config');
+    ($RcManager, $mode) = dns_mgr();
+    die 'wrong DNS-Manager is currently used for default' if ($RcManager !~ /symlink/ || $mode !~ /default/);
+    ping_check;
+    record_info('chronyd service check');
+    assert_script_run('chronyc -n sources');
+    assert_script_run(
+'curl https://git.tuxfamily.org/chrony/chrony.git/plain/examples/chrony.nm-dispatcher.dhcp -o /etc/NetworkManager/dispatcher.d/chrony.nm-dispatcher.dhcp'
+    );
+    script_run(
+        'cat > /etc/chrony.d/dhcp-source.conf << EOF
+sourcedir /var/run/chrony-dhcp
+EOF
+true'
+    );
+    systemctl('restart chronyd');
+    systemctl('restart NetworkManager');
```

Comment:
Since https://build.opensuse.org/request/show/1038940 it should be there in ALP without extra work.
Better do not install it via this hook, so we test the `chrony` package here as well.
```suggestion

```